### DISCUSSION TITLE
How to work with polygon custom control?

### DISCUSSION THREAD

**Discussion by Tennyleaz** [fabricjs/fabric.js#9675] [Q&A] [answered]
Hi,
I'm using fabric 5.3.0, and following the custom control tutorial in official website: [http://fabricjs.com/custom-controls-polygon](http://fabricjs.com/custom-controls-polygon).

My code snippets is like:
```javascript
let poly: fabric.Polygon = ...
let lastControl = poly.points.length - 1;
poly.controls = poly.points.reduce(function (
    activeContorls: { [key: string]: fabric.Control },
    point: fabric.Point,
    index: number
) {
    activeContorls["p" + index] = new fabric.Control({
        positionHandler: myPolyPositionHandler,
        actionHandler: myAnchorWrapper(index > 0 ? index - 1 : lastControl, myActionHandler),
        actionName: "modifyPolygon",
        pointIndex: index,
        render: myRenderAnchorControl,
    });
    return activeContorls;
},
    {});
```

My question is, how do I fire a custom event while start modifying a polygon, ex: `object:myModifyPolygon`, just like regular `object:scaling` or `object:moving` events from default object controls?
Do I need to manually fire something in actionHandler?

**Comment by ShaMan123** [comment, marked as answer]
Manually yes
There is wrapWithFireEvent, look at it
